assets/js/data/utils/process-error-response.ts

```diff
@@ -0,0 +1,141 @@
+/**
+ * External dependencies
+ */
+import {
+	createNotice,
+	createNoticeIfVisible,
+	DEFAULT_ERROR_MESSAGE,
+} from '@woocommerce/base-utils';
+import { decodeEntities } from '@wordpress/html-entities';
+import { isObject, objectHasProp, ApiErrorResponse } from '@woocommerce/types';
+import { noticeContexts } from '@woocommerce/base-context/event-emit/utils';
+
+type ApiParamError = {
+	param: string;
+	id: string;
+	code: string;
+	message: string;
+};
+
+const isApiResponse = ( response: unknown ): response is ApiErrorResponse => {
+	return (
+		isObject( response ) &&
+		objectHasProp( response, 'code' ) &&
+		objectHasProp( response, 'message' ) &&
+		objectHasProp( response, 'data' )
+	);
+};
+
+/**
+ * Flattens error details which are returned from the API when multiple params are not valid.
+ *
+ * - Codes will be prefixed with the param. For example, `invalid_email` becomes `billing_address_invalid_email`.
+ * - Additional error messages will be flattened alongside the main error message.
+ * - Supports 1 level of nesting.
+ * - Decodes HTML entities in error messages.
+ */
+const getErrorDetails = ( response: ApiErrorResponse ): ApiParamError[] => {
```

Comment:
This formats and flattens errors from our API schema into something more usable.